bloomberg/xcdiff
# Include target attributes in the attributes comparator

Resolves https://github.com/bloomberg/xcdiff/issues/62

**Overview**

- Target attributes are now included in the comparison results of the attributes comparator
- Each target will now have a dedicated comparison result 

e.g.

```
❌ ATTRIBUTES > "MyTarget" target

⚠️  Only in second (1):

  • LastSwiftMigration = 1110
```

- Target attributes may contain references to other targets, to ensure a semantic comparison is performed a target lookup is performed on attributes and resolved to their names if necessary
- A typical example is when setting a target application for a UI test target which sets the `TestTargetID` attribute

```
TargetAttributes = {
    3704EABFFE3515E47749586D = {
        CreatedOnToolsVersion = 12.5;
        TestTargetID = 76EC5DF0EEA681734B20EC5C; /* <--- Reference to another target */
    };
    76EC5DF0EEA681734B20EC5C = {
        CreatedOnToolsVersion = 12.4;
    };
};
``` 

- An additional fixture was added to contain projects with UI test targets which have target applications set. The underlying raw references are not identical, but are semantically equivalent (`MyAppUITests` > `MyApp`) 

**Test Plan**

- Run xcdiff on the fixtures
```sh
cd Fixtures
swift run xcdiff -g attributes -v
```
- Verify target attributes are now included in the results
- Run xcdiff on the new ui test fixtures 
```sh
cd Fixtures/ios_project_with_ui_tests
swift run xcdiff -g attributes -v
```
- Verify the `TestTargetID` attribute is not flagged as a difference
